### PR TITLE
ELECTRON-928 - Migrate Presence to the Power Monitor API

### PR DESCRIPTION
**Pull request by vrbsm**
## Description
 Blackrock Prod - Electron Desktop Presence in Citrix VDI environment not accurate
 [ELECTRON-928](https://perzoinc.atlassian.net/browse/ELECTRON-928).

## Related PRs
List related PRs against other branches / repositories:

branch | PR
------ | ------
ELECTRON-928-master | [link](https://github.com/SymphonyOSF/SFE-Client-App/pull/12974)

## QA Checklist
- [x] Unit-Tests
- [ ] Automation-Tests

## Blog Posts
https://github.com/facebook/jest/issues/936#issuecomment-214939935 

## Unit Tests
[Symphony Electron Test Result - ELECTRON-928.pdf](https://github.com/symphonyoss/SymphonyElectron/files/2657252/Symphony.Electron.Test.Result.-.ELECTRON-928.pdf)

### CODE REVIEW

File: js/activityDetection/index.js [hunk 1]
```diff
@@ -77,7 +77,7 @@ function monitorUserActivity() {
  */
 function sendActivity() {
     let systemActivity = activityDetection();
-    if (systemActivity && !systemActivity.isUserIdle && systemActivity.systemIdleTime) {
+    if (systemActivity && !systemActivity.isUserIdle && systemActivity.systemIdleTime != null) {
```
[the remark this answers is not in the document]
Comment (VishwasShashidhar) [reply js/activityDetection/index.js]: +1 @vrbsm please elaborate on how this will address the issue.

**Comment by VishwasShashidhar** [comment]
@vrbsm Let us migrate to https://github.com/electron/electron/blob/v4.0.0-beta.7/docs/api/power-monitor.md#powermonitorquerysystemidletimecallback

**Comment by KiranNiranjan** [comment]
@vrbsm, In my opinion, I think It's better to use `app.isReady()` and use the callback from the `powerMonitor.querySystemIdleTime` method to carry out the logic instead of an event emitter

```
if (app.isReady()) {
        powerMonitor.querySystemIdleTime((idleTime) => {
            if (idleTime < maxIdleTime) {
                send({ systemIdleTime: systemActivity.systemIdleTime });
            }
        });
}
```

File: js/activityDetection/index.js [hunk 2]
```diff
@@ -57,8 +74,9 @@ function initiateActivityDetection() {
 function monitorUserActivity() {
     intervalId = setInterval(monitor, 1000);
 
-    function monitor() {
-        if (systemIdleTime.getIdleTime() < maxIdleTime && typeof getIsOnlineFnc === 'function' && getIsOnlineFnc()) {
+    async function monitor() {
+        let idleTime = await getIdleTime();
+        if (idleTime < maxIdleTime && typeof getIsOnlineFnc === 'function' && getIsOnlineFnc()) {
```
[the remark this answers is not in the document]
Comment (vrbsm) [reply js/activityDetection/index.js]: Fixed

**Comment by VishwasShashidhar** [comment]
@vrbsm looks good. but, please also attach the unit tests results.

**Comment by VishwasShashidhar** [comment]
@vrbsm Please remove dependency on `@paulcbetts/system-idle-time` in `package.json` and push the changes. I'll merge the PR.

**Pull request merged**